src/schema/city/index.ts

```diff
@@ -53,31 +54,57 @@ const CityType = new GraphQLObjectType<any, ResolverContext>({
         },
       }),
       resolve: async (city, args, { showsWithHeadersLoader }) => {
-        // default Enum value for status is not properly resolved
-        // so we have to manually resolve it by lowercasing the value
-        // https://github.com/apollographql/graphql-tools/issues/715
-        args.status = args.status.toLowerCase()
-
-        const gravityOptions = {
-          ...convertConnectionArgsToGravityArgs(args),
-          displayable: true,
+        const baseParams = {
           near: `${city.coordinates.lat},${city.coordinates.lng}`,
           max_distance: LOCAL_DISCOVERY_RADIUS_KM,
           has_location: true,
-          total_count: true,
+          sort: args.sort,
+          // default Enum value for status is not properly resolved
+          // so we have to manually resolve it by lowercasing the value
+          // https://github.com/apollographql/graphql-tools/issues/715
+          status: args.status.toLowerCase(),
+          // This is to ensure the key never makes its way into the `baseParams`
+          // object if not needed,
+          ...(typeof args.discoverable === "undefined"
+            ? undefined
+            : { discoverable: args.discoverable }),
+          ...(typeof args.discoverable === "undefined"
+            ? { displayable: true }
+            : undefined),
         }
-        delete gravityOptions.page
 
-        if (args.discoverable) {
-          delete gravityOptions.displayable
+        let response
+        let offset
+
+        if (args.first === MAX_GRAPHQL_INT) {
```

Comment:
im not sure if i understand this correctly, but this means we make one big query to get everything in one request, is it possible to cache things with pagination? basically set page size to 100 and make multiple requests asynch till we fetch everything if there are more than 100 items.

Reply:
No it uses the `allViaLoader`, which does what you suggest but in batches of 25.